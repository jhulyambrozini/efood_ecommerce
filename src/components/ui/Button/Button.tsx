import { ButtonContainer } from './styles'

export type ButtonProps = {
  type: 'button' | 'submit'
  background: 'primary' | 'secundary'
  text?: string
  title: string
  onClick?: React.MouseEventHandler<HTMLButtonElement>
}

const Button = ({ type, background, text, title, onClick }: ButtonProps) => (
  <ButtonContainer
    background={background}
    title={title}
    type={type}
    onClick={onClick}
  >
    {text ? text : title}
  </ButtonContainer>
)

export default Button
