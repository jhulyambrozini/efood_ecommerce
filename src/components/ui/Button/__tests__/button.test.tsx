import '@testing-library/jest-dom'
import { vi } from 'vitest'
import { fireEvent, render, screen } from '@testing-library/react'

import Button from '..'

describe('<Button />', () => {
  it('should render a primary button"', () => {
    render(
      <Button
        label="button primary"
        background="primary"
        type="button"
        title="click me"
      />
    )
    const buttonPrimary = screen.getByRole('button', {
      name: /button primary/i
    })
    expect(buttonPrimary).toHaveStyle('color:  rgb(255, 235, 217)')
  })

  it('should render a secondary button"', () => {
    render(
      <Button
        label="button secondary"
        background="secundary"
        type="button"
        title="click me"
      />
    )
    const buttonSecondary = screen.getByRole('button', {
      name: /button secondary/i
    })
    expect(buttonSecondary).toHaveStyle('color:   rgb(177, 57, 57)')
  })

  it('should call function on button click', () => {
    const handleClick = vi.fn()
    render(
      <Button
        label="button"
        background="secundary"
        type="button"
        title="click me"
        onClick={handleClick}
      />
    )
    const button = screen.getByRole('button', {
      name: /button/i
    })
    fireEvent.click(button)

    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('should match snapshot', () => {
    const handleClick = vi.fn()

    const { container } = render(
      <Button
        label="button"
        background="secundary"
        type="button"
        title="click me"
        onClick={handleClick}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
