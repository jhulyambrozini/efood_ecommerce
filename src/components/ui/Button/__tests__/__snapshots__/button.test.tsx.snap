// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Button /> > should match snapshot 1`] = `
<button
  class="sc-dcJsrY bDWePn"
  title="click me"
  type="button"
>
  button
</button>
`;

exports[`<Button /> should match snapshot 1`] = `
<button
  class="sc-dcJsrY bDWePn"
  title="click me"
  type="button"
>
  button
</button>
`;
